src/ophyd_async/epics/adandor/_andor_controller.py

```diff
@@ -0,0 +1,49 @@
+import asyncio
+
+from ophyd_async.core import (
+    DetectorTrigger,
+    TriggerInfo,
+)
+from ophyd_async.epics import adcore
+
+from ._andor_io import Andor2DriverIO, Andor2TriggerMode, ImageMode
+
+_MIN_DEAD_TIME = 0.1
+_MAX_NUM_IMAGE = 999_999
+
+
+class Andor2Controller(adcore.ADBaseController[Andor2DriverIO]):
+    def __init__(
+        self,
+        driver: Andor2DriverIO,
+        good_states: frozenset[adcore.DetectorState] = adcore.DEFAULT_GOOD_STATES,
+    ) -> None:
+        super().__init__(driver, good_states=good_states)
+
+    def get_deadtime(self, exposure: float | None) -> float:
+        return _MIN_DEAD_TIME + (exposure or 0)
+
+    async def prepare(self, trigger_info: TriggerInfo):
+        await self.set_exposure_time_and_acquire_period_if_supplied(
+            trigger_info.livetime
+        )
+        await asyncio.gather(
+            self.driver.trigger_mode.set(self._get_trigger_mode(trigger_info.trigger)),
+            self.driver.num_images.set(
+                trigger_info.total_number_of_triggers or _MAX_NUM_IMAGE
+            ),
+            self.driver.image_mode.set(ImageMode.MULTIPLE),
+        )
```

Comment:
Could we not subclass `TriggerInfo` like eiger does, making an `Andor2TriggerInfo` which contains `trigger: Andor2TriggerMode` instead of `trigger: DetectorTrigger`?

Reply:
I'm not sure I understand, the only eiger code I can find is:
https://github.com/bluesky/ophyd-async/blob/4f4458c916071859c44b03b2f39a6a9af05d4aac/src/ophyd_async/epics/eiger/_eiger_controller.py#L39-L53

Plus, the point of the controller class it to translate from the generic `TrigerInfo` to the specific `Andor2TriggerMode`, so this translation should definitely happen here...

Reply:
Ah, that was so that Eiger can add something extra, `energy`, that can't be determined from what's in `TriggerInfo`, while `Andor2TriggerMode` can already be determined from a standard `TriggerMode`. I'm not convinced we need `EigerTriggerMode` now that we have intelligent load either, so opened https://github.com/bluesky/ophyd-async/issues/727 to discuss